tutorials/provisional/mosaic.py

```diff
@@ -0,0 +1,295 @@
+        title to be equal to the key.
+    """
+    for k, ax in ax_dict.items():
+        ax.set_title(k, loc="left")
+
+
+###############################################################################
+# String specification
+# ====================
+#
+# Using single letters we can "draw" the Axes we want.  The following
+
+
+layout = """
+AB
+CD
+"""
+
+###############################################################################
+# will give us 4 Axes laid out in an 2x2 grid and generates the
+# same figure layout as ``fig.subplots(2, 2)``.
+
+fig = plt.figure(constrained_layout=True)
+axd = fig.subplot_mosaic(layout)
+label_axes(axd)
+
+###############################################################################
+# A key difference between `.Figure.subplots` and
+# `.Figure.subplot_mosaic` is that the return value from
+# the latter is a dictionary mapping the labels to the
+# `.axes.Axes` instances created
+
+print(axd)
+
+###############################################################################
+# instead of an array.  For comparison we could do:
+
+fig = plt.figure(constrained_layout=True)
+ax_array = fig.subplots(2, 2, squeeze=False)
+label_axes({str((j, k)): ax
+            for j, row in enumerate(ax_array)
+            for k, ax in enumerate(row)})
+
+###############################################################################
+# If we wanted to use the upper left Axes we would do either
+
+upper_left = axd['A']
+upper_left = ax_array[0, 0]
+
+
+###############################################################################
+# Something else we can easily do with `.Figure.subplot_mosaic` is
+# specify a span.  If we want to add an Axes to the right side of
+# our Figure that spans the full vertical space we could do
+
+axd = plt.figure(constrained_layout=True).subplot_mosaic(
+    """
+    ABE
+    CDE
+    """
+)
+label_axes(axd)
+
+###############################################################################
+# If we do not want to fill in all the spaces in the Figure with Axes,
+# we can specify it to be blank
+
+
+axd = plt.figure(constrained_layout=True).subplot_mosaic(
+    """
+a.c
+BBB
+.d.
+"""
+)
+label_axes(axd)
+
+
+###############################################################################
+# If we prefer to use another character (rather than a period ``"."``)
+# to mark the empty space, we can use *empty_sentinel* to specify the
+# character to use.
+
+axd = plt.figure(constrained_layout=True).subplot_mosaic(
+    """
+    aX
+    Xb
+    """,
+    empty_sentinel="X",
+)
+label_axes(axd)
+
+
+###############################################################################
+#
+# Because these are string inputs, any Unicode glyph is valid!
+
+axd = plt.figure(constrained_layout=True).subplot_mosaic(
+    """αб
+       ℝ☢"""
+)
+label_axes(axd)
+
+###############################################################################
+# It is not recommended to use white space as either a label or an empty
```

Comment:
Better to just forbid pure white space, I think, and always strip strings.

Reply:
In the nested-list format there isn't a problem and I'm not sure that it is worth writing the logic to forbid a user doing 

```python
layout = """
a a
bbb
"""
```

Reply:
Certainly, we can leave it as-is and see what happens.